internal/ethapi/api.go

```diff
@@ -80,7 +80,7 @@ func (s *PublicEthereumAPI) Syncing() (interface{}, error) {
 	progress := s.b.Downloader().Progress()
 
 	// Return not syncing if the synchronisation already completed
-	if progress.CurrentBlock >= progress.HighestBlock {
+	if !s.b.Downloader().Synchronising() {
```

Comment:
This won't be a good change:

```
// Synchronising returns whether the downloader is currently retrieving blocks.
func (d *Downloader) Synchronising() bool {
	return atomic.LoadInt32(&d.synchronising) > 0
}
```

During the node's initial sync (and maybe even afterwards), sync will abort and restart a few times if the master peer we're connected to disappears. The current code `progress.CurrentBlock >= progress.HighestBlock` doesn't care about that, and will just report syncing until we're actually in sync. Your change would report sync done whenever the downloader aborts for a few second.